File: src/logger.ts
import { invoke, isTauri } from "@tauri-apps/api/core";

export default class Logger {
    public static warn(scope: string, message: string) {
        console.log(`[WARN][${scope}] ${message}`);
        if (isTauri())
            invoke("logger_warn", { scope: `Web - ${scope}`, message });
    }
    public static info(scope: string, message: string) {
        console.log(`[INFO][${scope}] ${message}`);
        if (isTauri())
            invoke("logger_info", { scope: `Web - ${scope}`, message });
    }
    public static error(scope: string, message: string) {
        console.log(`[ERROR][${scope}] ${message}`);
        if (isTauri())
            invoke("logger_error", { scope: `Web - ${scope}`, message });
    }
    public static fatal(scope: string, message: string) {
        console.log(`[FATAL][${scope}] ${message}`);
        if (isTauri())
            invoke("logger_fatal", { scope: `Web - ${scope}`, message });
    }
}

File: src-tauri/src/logger.rs
pub struct Logger;

impl Logger {
    pub fn info<T>(scope: T, message: T) where T : ToString {
        println!("[INFO][{}] {}", scope.to_string(), message.to_string());
    }
    pub fn warn<T>(scope: T, message: T) where T : ToString {
        println!("[WARN][{}] {}", scope.to_string(), message.to_string());
    }
    pub fn error<T>(scope: T, message: T) where T : ToString {
        println!("[ERROR][{}] {}", scope.to_string(), message.to_string());
    }
    pub fn fatal<T>(scope: T, message: T) where T : ToString {
        println!("[FATAL][{}] {}", scope.to_string(), message.to_string());
    }
}
